src/main/performanceMonitor.ts

```diff
@@ -0,0 +1,171 @@
+
+        webContents.on('did-finish-load', () => {
+            this.serverViews.set(webContents.id, {name, webContents, serverId});
+        });
+    };
+
+    unregisterView = (webContentsId: number) => {
+        log.debug('unregisterView', webContentsId);
+
+        this.views.delete(webContentsId);
+        this.serverViews.delete(webContentsId);
+    };
+
+    private start = () => {
+        if (!this.isInitted) {
+            return;
+        }
+
+        if (!Config.enableMetrics) {
+            return;
+        }
+
+        log.verbose('start');
+
+        if (this.updateInterval) {
+            clearInterval(this.updateInterval);
+        }
+        this.updateInterval = setInterval(this.sendMetrics, METRIC_SEND_INTERVAL);
+    };
+
+    private stop = () => {
+        log.verbose('stop');
+
+        clearInterval(this.updateInterval);
+        delete this.updateInterval;
+    };
+
+    private runMetrics = async () => {
+        const metricsMap: Map<string, Metrics> = new Map();
+
+        // Collect metrics for all of the Node processes
+        app.getAppMetrics().
+            filter((metric) => metric.type !== 'Tab').
+            forEach((metric) => {
+                metricsMap.set(metric.name ?? metric.type, {
+                    cpu: metric.cpu.percentCPUUsage,
+                    memory: metric.memory.privateBytes ?? metric.memory.workingSetSize,
+                });
+            });
+
+        const viewResolves: Map<number, () => void> = new Map();
+        const listener = (event: IpcMainEvent, name: string, metrics: Metrics) => {
+            metricsMap.set(name, metrics);
+            viewResolves.get(event.sender.id)?.();
+        };
+        ipcMain.on(METRICS_RECEIVE, listener);
+        const viewPromises = [...this.views.values(), ...this.serverViews.values()].map((view) => {
+            return new Promise<void>((resolve) => {
+                viewResolves.set(view.webContents.id, resolve);
+                view.webContents.send(METRICS_REQUEST, view.name, view.serverId);
+            });
+        });
+        await Promise.all(viewPromises);
+        ipcMain.off(METRICS_RECEIVE, listener);
+        return metricsMap;
+    };
+
+    private sendMetrics = async () => {
+        const metricsMap = await this.runMetrics();
+        for (const view of this.serverViews.values()) {
+            const serverId = view.serverId;
+            if (!serverId) {
+                log.error(`Cannot send metrics for ${view.name}  - missing server id`);
+                continue;
+            }
+
+            if (!view.webContents) {
+                log.error(`Cannot send metrics for ${view.name}  - missing web contents`);
+                continue;
+            }
+
+            const serverMetricsMap = new Map([...metricsMap].filter((value) => !value[1].serverId || value[1].serverId === view.serverId));
+            view.webContents?.send(METRICS_SEND, serverMetricsMap);
```

Comment:
no need for optional chaining `view.webContents?.send` since it occurs after checking `if (!view.webContents)`